File: java/parsers/src/test/java/jp/co/teruuu/parser/common/EOFParserTest.java
package jp.co.teruuu.parser.common;

import jp.co.teruuu.parser.common.ParseResult;
import jp.co.teruuu.parser.common.Parser;
import jp.co.teruuu.parser.common.type.Tuple2;
import org.junit.Test;

import java.util.Arrays;
import java.util.List;

public class EOFParserTest {

    @Test
    public void test(){
        Parser<Tuple2<List<String>, String>> parser = Parser.string("Hello").many().pair2(Parser.EOF());

        ParseResult result1 = parser.parse("HelloHelloHello");
        assert(result1 instanceof ParseResult.Success);
        assert(((ParseResult.Success<Tuple2<List<String>, String>>) result1).value.item1.equals(Arrays.asList("Hello", "Hello","Hello")));
        assert(((ParseResult.Success<Tuple2<List<String>, String>>) result1).value.item2.equals(""));

        ParseResult result2 = parser.parse("Hell oHelloHello");
        assert(result2 instanceof ParseResult.Failure);


    }
}

File: scala/parsers/src/test/scala/jp/co/teruuu/parser/common/EOFParserTest.scala
package jp.co.teruuu.parser.common

import org.scalatest.FlatSpec;

class EOFParserTest extends FlatSpec {

  "parse" should "successt" in {
    val parser = Parser.eof
    val result1 = parser.parse("")
    assert(result1.isInstanceOf[ParseOk[_]])
  }

  "parse" should "failer" in {
    val parser = Parser.eof
    val result1 = parser.parse("Hello")
    assert(result1.isInstanceOf[ParseNg[_]])
  }
}
